File: FoodDelivery.Entities/FilterParams/DishFilterParams.cs
using System.Collections.Generic;
using FoodDelivery.Entities.Enums.Sorts;

namespace FoodDelivery.Entities.FilterParams
{
	public class DishFilterParams : BaseFilterParams
	{
		public List<string> Restaurants { get; set; }
		public List<string> Categories { get; set; }
		public decimal? MinPrice { get; set; }
		public decimal? MaxPrice { get; set; }
		public DishSortType DishSortType { get; set; }
	}
}

File: FoodDelivery.Web/src/app/models/filters/dishFilterParams.ts
import { DishSortType } from '../enums/sorts/DishSortType';
import { BaseFilterParams } from './BaseFilterParams';

export class DishFilterParams extends BaseFilterParams{
    public restaurants? : string[];
    public categories? : string[];
    public dishSortType: DishSortType;
    public minPrice: number;
    public maxPrice: number;
}